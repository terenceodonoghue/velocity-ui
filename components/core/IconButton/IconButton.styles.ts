import { css, FunctionInterpolation, Theme } from '@emotion/react';
import { em, rgba } from 'polished';

const iconButton: FunctionInterpolation<Theme> = ({
  palette,
  transitions,
  typography,
}) => css`
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  border-radius: 5px;
  padding: 4px;
  background: none;
  font-size: ${em(15, typography.fontSize)};
  font-weight: ${typography.fontWeight.medium};
  cursor: pointer;
  transition: background-color ${transitions.duration.standard}ms
    ${transitions.easing.easeInOut};

  &:disabled {
    opacity: 0.3;
    pointer-events: none;
  }

  &:focus {
    outline: 0;
  }

  &:enabled:hover {
    background-color: ${rgba(palette.accent, 0.1)};
  }
`;

export default iconButton;
