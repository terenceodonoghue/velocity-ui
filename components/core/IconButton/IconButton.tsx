import React, { ButtonHTMLAttributes, FunctionComponent } from 'react';
import iconButton from './IconButton.styles';

const IconButton: FunctionComponent<ButtonHTMLAttributes<HTMLButtonElement>> = (
  props,
) => (
  <button css={iconButton} data-testid="iconbutton" type="button" {...props} />
);

export default IconButton;
