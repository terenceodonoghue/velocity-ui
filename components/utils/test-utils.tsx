import React, { FunctionComponent, ReactElement } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import ThemeProvider from '~/components/core/ThemeProvider';

const Providers: FunctionComponent = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult => render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

export { customRender as render };
