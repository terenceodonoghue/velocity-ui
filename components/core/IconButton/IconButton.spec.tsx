import React from 'react';
import { render } from 'test-utils';
import IconButton from './IconButton';

describe('IconButton', () => {
  it('renders button', () => {
    const { container } = render(<IconButton />);

    // toContainElement(testid)
    expect(container).toMatchSnapshot();
  });
});
