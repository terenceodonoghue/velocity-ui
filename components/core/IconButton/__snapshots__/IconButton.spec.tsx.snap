// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton renders button 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: none;
  border-radius: 5px;
  padding: 4px;
  background: none;
  font-size: 1em;
  font-weight: 500;
  cursor: pointer;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1);
}

.emotion-0:disabled {
  opacity: 0.3;
  pointer-events: none;
}

.emotion-0:focus {
  outline: 0;
}

.emotion-0:enabled:hover {
  background-color: rgba(46,91,255,0.1);
}

<div>
  <button
    class="emotion-0"
    data-testid="iconbutton"
    type="button"
  />
</div>
`;
